pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/generic/GenericJsonSchema.java

```diff
@@ -43,4 +51,88 @@ public GenericJsonSchema(SchemaInfo schemaInfo) {
     public GenericRecordBuilder newRecordBuilder() {
         throw new UnsupportedOperationException("Json Schema doesn't support record builder yet");
     }
+
+    static org.apache.pulsar.client.api.Schema<?> convertFieldSchema(JsonNode fn) {
+        if (fn.isContainerNode()) {
+            return buildStructSchema(fn, new AtomicInteger());
+        }
+        switch (fn.getNodeType()) {
+            case STRING:
+                return GenericSchema.STRING;
+            case BOOLEAN:
+                return GenericSchema.BOOL;
+            case BINARY:
+                return GenericSchema.BYTES;
+            case NUMBER:
+                if (fn.isInt()) {
+                    return GenericSchema.INT32;
+                } else if (fn.isLong()) {
+                    return GenericSchema.INT64;
+                } else if (fn.isFloat()) {
+                    return GenericSchema.FLOAT;
+                } else if (fn.isDouble()) {
+                    return GenericSchema.DOUBLE;
+                } else {
+                    return null;
```

Comment:
(here and in other places) Consider using `Optional<>`. 
Are you using nulls for the performance reasons? how frequently this code runs?